tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodAttributeExecuteIntegration.cs

```diff
@@ -56,80 +74,99 @@ public static class TestMethodAttributeExecuteIntegration
-
-                    if (testResult.TestFailureException != null)
+                    var test = i == 0 ?
+                                   testMethodState.Test :
+                                   MsTestIntegration.OnMethodBegin(testMethodState.TestMethod, testMethodState.TestMethod.Type, testMethodState.Test.StartTime);
+                    var testResultObject = returnValueList[i];
+                    if (testResultObject.TryDuckCast<TestResultStruct>(out var testResult))
                     {
-                        var testException = testResult.TestFailureException.InnerException ?? testResult.TestFailureException;
-                        var testExceptionType = testException.GetType();
-                        var testExceptionName = testExceptionType.Name;
-                        if (testExceptionName != "UnitTestAssertException" && testExceptionName != "AssertInconclusiveException")
+                        var testException = testResult.TestFailureException?.InnerException ??
+                                            testResult.TestFailureException ??
+                                            exception;
+                        if (testException != null)
                         {
-                            test.SetErrorInfo(testException);
-                        }
-
-                        errorType = testExceptionType.FullName;
-                        errorMessage = testException.Message;
-                        errorStackTrace = testException.ToString();
-                    }
-
-                    switch (testResult.Outcome)
-                    {
-                        case UnitTestOutcome.Error:
-                        case UnitTestOutcome.Failed:
-                        case UnitTestOutcome.Timeout:
-                            test.SetErrorInfo(errorType, errorMessage, errorStackTrace);
-                            test.Close(TestStatus.Fail);
-                            break;
-                        case UnitTestOutcome.Inconclusive:
-                        case UnitTestOutcome.NotRunnable:
-                            if (exception is not null)
+                            var testExceptionType = testException.GetType();
+                            var testExceptionName = testExceptionType.Name;
+                            if (testExceptionName != "UnitTestAssertException" && testExceptionName != "AssertInconclusiveException")
                             {
-                                test.SetErrorInfo(exception);
-                                test.Close(TestStatus.Fail);
+                                test.SetErrorInfo(testException);
                             }
                             else
                             {
-                                test.Close(TestStatus.Skip, TimeSpan.Zero, errorMessage);
+                                test.SetErrorInfo(testExceptionType.FullName ?? "Error", testException.Message, testException.ToString());
```

Comment:
Note that `span.SetException()` uses `testExceptionType.ToString()` to set the exception type, not `testExceptionType.FullName`, and they have [slightly different behaviours](https://learn.microsoft.com/en-us/dotnet/api/system.type.tostring?view=net-8.0). Maybe you want to use `ToString() for consistency?

Reply:
oh you're right, changed in [f4d4b7a](https://github.com/DataDog/dd-trace-dotnet/pull/5381/commits/f4d4b7acfdfab96cfbb9680c8ef1299db40978f7)